src/EventSubscriber/DefaultVariableSubscriber.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Drupal\wmcontent\EventSubscriber;
+
+use Drupal\core_event_dispatcher\Event\Theme\TemplatePreprocessDefaultVariablesAlterEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class DefaultVariableSubscriber implements EventSubscriberInterface
+{
+    /** @var RequestStack */
+    protected $requestStack;
+
+    public function __construct(
+        RequestStack $requestStack,
+    ) {
+        $this->requestStack = $requestStack;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'hook_event_dispatcher.theme.template_preprocess_default_variables_alter' => 'alter',
+        ];
+    }
+
+    public function alter(TemplatePreprocessDefaultVariablesAlterEvent $event)
+    {
+        $variables = &$event->getVariables();
+        $request = $this->requestStack->getCurrentRequest();
+
+        $variables['is_preview'] = ($request?->query->get('preview', 'false') === 'true');
```

Comment:
- Can you move this to the `wmcontent.module` file? `wmcontent` doesn't have a dependency on `hook_event_dispatcher` and I'd like to avoid adding it.
    - See [user_template_preprocess_default_variables_alter in user.module](https://git.drupalcode.org/project/drupal/-/blob/10.2.2/core/modules/user/user.module?ref_type=tags#L350-359) for inspiration
- Can you namespace your twig variable and query parameter? It's very well possible `is_preview` is already used somewhere, or some websites have need that twig or query parameter for another feature.
    - Eg `wmcontent_is_preview`